packages/components/src/button/test/index.js

```diff
@@ -75,6 +75,29 @@ describe( 'Button', () => {
 			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-pressed' );
 		} );
 
+		it( 'should render a button element with has-text when children are passed', async () => {
+			render( <Button icon={ plusCircle }>Children</Button> );
+			expect( screen.getByRole( 'button' ) ).toHaveClass( 'has-text' );
+		} );
+
+		it( 'should render a button element without has-text when children are not passed', async () => {
+			render( <Button icon={ plusCircle }></Button> );
+			expect( screen.getByRole( 'button' ) ).not.toHaveClass(
+				'has-text'
+			);
+		} );
+
+		it( 'should render a button element without has-text when children are empty flagment', async () => {
+			render(
+				<Button icon={ plusCircle }>
+					<></>
+				</Button>
+			);
+			expect( screen.getByRole( 'button' ) ).not.toHaveClass(
+				'has-text'
```

Comment:
Eventually I'm hoping we can move these kinds of tests to visual regression testing because they are more robust than checking for class names ✨